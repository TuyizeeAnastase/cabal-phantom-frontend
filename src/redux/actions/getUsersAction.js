import { GET_USER_REQUEST, GET_USER_SUCCESS, GET_USER_FAIL } from '../types/getUsersType'
import axios from 'axios'

export const getUsersAction = () => async (dispatch) => {
    try {
        dispatch({ type: GET_USER_REQUEST })
        const token = localStorage.getItem('token')

        const res = await axios.get('https://phantom-cabal-staging.herokuapp.com/api/v1/auth', {
            headers: {
                'Authorization': token
            }
        });
        
        return dispatch(getUsersSuccess(res))
        
    } catch (err) {
        if (error.response) {
            console.log(err);
            return dispatch(getUsersFail('Network Fails'))
        }
    }
}

export const getUsersSuccess = data => {
    return {
        type: GET_USER_SUCCESS,
        payload: data
    }
}

export const getUsersFail = error => {
    return {
        type: GET_USER_FAIL,
        payload: error
    }
}
