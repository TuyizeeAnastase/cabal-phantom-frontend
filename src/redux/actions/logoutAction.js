import {LOGOUT_REQUEST, LOGOUT_SUCCESS, LOGOUT_ERROR} from '../types/logoutTypes'
import axios from 'axios'

export const logout = () => async(dispatch) => {
    try {
        dispatch(logoutRequest());
        // console.log('Request')
        const res = await axios.get('http://localhost:5000/api/v1/auth/logout')
        localStorage.removeItem('token')
        localStorage.clear()
        dispatch(logoutSuccess(res))
        console.log('Response', res)

    } catch (error) {
        console.log(error)
        dispatch(logoutError(error))
    }    
};
    
export const logoutRequest = () =>{
    return { type: LOGOUT_REQUEST }
}  

export const logoutSuccess = (data) =>{
    return { 
        type: LOGOUT_SUCCESS,
        payload: data
    }
}  

export const logoutError = (error) =>{
    return { 
        type: LOGOUT_ERROR,
        payload: error
    }
}  
