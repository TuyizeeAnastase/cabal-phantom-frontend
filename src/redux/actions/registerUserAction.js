import { REGISTER_REQUEST, REGISTER_SUCCESS, REGISTER_FAIL } from '../types/registerUsersTypes';
import axios from 'axios';

export const registerUserAction = (data) => async (dispatch) => {
    try {
        dispatch({ type: REGISTER_REQUEST });
        const token = localStorage.getItem('token')

        const res = await axios.post(
            'https://phantom-cabal-staging.herokuapp.com/api/v1/auth/register', data, {
            headers: {
                'Authorization': token
            }
        });
        console.log('Action:\n', res)
        return dispatch(registerSuccess(res));
    } catch (error) {
        if (error.response) {
            const errorMessage = await error.response.data.message;
            return dispatch(registerFails(errorMessage));
        } 
        else {
            console.log(error)
            return dispatch(registerFails('Network Error'));
        }
    }
};

export const registerSuccess = (data) => {
    return {
        type: REGISTER_SUCCESS,
        payload: data,
    };
};

export const registerFails = (error) => {
    return {
        type: REGISTER_FAIL,
        payload: error,
    };
};