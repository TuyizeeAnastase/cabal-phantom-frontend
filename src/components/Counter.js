import React, { Component } from 'react';
import { increment, decrement } from '../redux/actions/testAction'
import { connect } from "react-redux";
import AddIcon from '@material-ui/icons/Add';
import RemoveIcon from '@material-ui/icons/Remove';
import '../Assets/Styles/App.scss'

class Counter extends Component {
    render() {
        return (
            <div className="headers">
                <h1 className="headers__title">Phantom App</h1>
                <div className="headers__title" aria-label="value">{this.props.value}</div>
                <button onClick={() => { this.props.dispatch(increment()) }} aria-label="incr"><AddIcon style={{ fontSize: "40px" }} /></button>
                <button onClick={() => { this.props.dispatch(decrement()) }} aria-label="decr"><RemoveIcon style={{ fontSize: "40px" }} /></button>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        value: state.testRedux.value,
    };
};

export default connect(mapStateToProps)(Counter);
