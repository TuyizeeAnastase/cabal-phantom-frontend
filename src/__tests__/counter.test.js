import React from 'react'
import '@testing-library/jest-dom'
import { render } from "@testing-library/react";
import counterReducer from '../redux/reducers/testReducer';
import { increment, decrement } from '../redux/actions/testAction';
import Counter from '../components/Counter';
import { Provider } from 'react-redux'
import store from '../redux/store'
import userEvent from '@testing-library/user-event'

const AppComponent = () => {
    return render(
        <Provider store={store}>
            <Counter />
        </Provider>
    )
}

describe('Test Counter', () => {
    it('Capturing Snapshot of Counter', () => {
        const renderedValue = AppComponent()
        expect(renderedValue).toMatchSnapshot();
    });

    it("should test testAction", () => {
        expect(increment()).toHaveProperty('type')
        expect(decrement()).toHaveProperty('type')
    })

    it('should increment in testReducer', () => {
        let state = { value: 0 }
        state = counterReducer(state, { type: "INCREMENT" })
        expect(state).toEqual({ value: 1 })
    });

    it('should decrement in testReducer', () => {
        let state = { value: 0 }
        state = counterReducer(state, { type: "DECREMENT" })
        expect(state).toEqual({ value: -1 })
    });

    it('should return state in testReducer', () => {
        let state = { value: 0 }
        state = counterReducer(state, { type: "" })
        expect(state).toEqual(state)
    });

    it("should increment", () => {
        const { getByLabelText } = AppComponent();
        const incrementButton = getByLabelText("incr")
        const value = getByLabelText("value")
        const num = parseInt(value.innerHTML);
        userEvent.click(incrementButton)
        expect(parseInt(value.innerHTML)).toBe(num + 1)
    })

    it("should decrement", () => {
        const { getByLabelText } = AppComponent();
        const decrementButton = getByLabelText("decr")
        const value = getByLabelText("value")
        const val = parseInt(value.innerHTML);
        userEvent.click(decrementButton)
        expect(parseInt(value.innerHTML)).toBe(val - 1)
    })

});
