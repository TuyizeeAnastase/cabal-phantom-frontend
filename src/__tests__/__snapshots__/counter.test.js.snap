// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Counter Capturing Snapshot of Counter 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="headers"
      >
        <h1
          class="headers__title"
        >
          Phantom App
        </h1>
        <div
          aria-label="value"
          class="headers__title"
        >
          0
        </div>
        <button
          aria-label="incr"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            style="font-size: 40px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </button>
        <button
          aria-label="decr"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            style="font-size: 40px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="headers"
    >
      <h1
        class="headers__title"
      >
        Phantom App
      </h1>
      <div
        aria-label="value"
        class="headers__title"
      >
        0
      </div>
      <button
        aria-label="incr"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          style="font-size: 40px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </button>
      <button
        aria-label="decr"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          style="font-size: 40px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13H5v-2h14v2z"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
